plenum/test/cli/helper.py

```diff
@@ -258,24 +258,40 @@ def newCLI(looper, basedir, cliClass=TestCli,
            config=None,
            partition: str=None,
            unique_name=None,
-           logFileName=None):
+           logFileName=None,
+           name=None,
+           agentCreator=None):
     if partition:
         recorder = Recorder(partition)
     else:
         recorder = CombinedRecorder()
     mockOutput = MockOutput(recorder=recorder)
     recorder.write("~ be {}\n".format(unique_name))
     otags = config.log_override_tags['cli'] if config else None
-    newcli = cliClass(looper=looper,
-                      basedirpath=basedir,
-                      nodeReg=None,
-                      cliNodeReg=None,
-                      output=mockOutput,
-                      debug=True,
-                      config=config,
-                      unique_name=unique_name,
-                      override_tags=otags,
-                      logFileName=logFileName)
+    if name is not None and agentCreator is not None:
```

Comment:
I tried that approach as well. However, we cannot pass extra parameters while creating class instance, unless class is expecting either `kwargs` or `args`. Also, if try to remove values from dictionary on the basis of if condition, then also it will remain almost same. I did not see much benefit in that approach as well.

However, let me change it.